### PR TITLE
*(ticdc):  remove cdc context

### LINKED ISSUE #10841: remove cdc context
The cdc context package enhances the context of Golang by implicitly adding two variables for convenient access by various layers of code,  these two variables are ChangefeedVars and GlobalVars. 
However, the initialization of these variables is quite chaotic and not easily understandable. Moreover, this method of passing values cannot be detected by the compiler, adding to the maintenance burden.
After multiple rounds of code refactoring, we no longer need the context package, remove this package will simplify the CDC code

### PR DESCRIPTION
**Pull request by sdojjy**
### What problem does this PR solve?

Issue Number: close #10841

### What is changed and how it works?

The CDCContext structure in TiCDC contains numerous hidden parameters, which complicates the code by obscuring the logic of parameter passing across different goroutines and functions. This lack of transparency makes the codebase difficult to read and maintain.

Solution in the PR:
The PR remove the CDCContext structure, eliminating the need to pass a complex context object through the code. Instead, passing necessary parameters explicitly between functions, which improves code clarity.

#### Tests 

 - Unit test
 - Integration test
 - Manual test (add detailed scripts or steps below)
 - No code

### Release note 

```release-note
`None`.
```

### CODE REVIEW

**Comment by asddongmen** [comment]
Please also change the package name from context to another more proper name.

**Comment by zhangjinpeng1987** [comment]
@sdojjy Please describe in the PR what problem you are trying to solve, and how do you solve it.

File: pkg/vars/vars.go [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+	"github.com/pingcap/tiflow/pkg/config"
+	"github.com/pingcap/tiflow/pkg/etcd"
+	"github.com/pingcap/tiflow/pkg/p2p"
+	"github.com/tikv/client-go/v2/oracle"
+)
+
+// GlobalVars contains some vars which can be used anywhere in a pipeline
+// the lifecycle of vars in the GlobalVars should be aligned with the ticdc server process.
+// All field in Vars should be READ-ONLY and THREAD-SAFE
+type GlobalVars struct {
```
Comment (3AceShowHand) [inline pkg/vars/vars.go:30]: How about just naming it as `Global`, it's simpler

File: pkg/vars/vars.go [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+
+	// MessageServer and MessageRouter are for peer-messaging
+	MessageServer *p2p.MessageServer
+	MessageRouter p2p.MessageRouter
+}
+
+// ChangefeedVars contains some vars which can be used anywhere in a pipeline
+// the lifecycle of vars in the ChangefeedVars should be aligned with the changefeed.
+// All field in Vars should be READ-ONLY and THREAD-SAFE
+type ChangefeedVars struct {
```
Comment (3AceShowHand) [inline pkg/vars/vars.go:48]: I doubt is this necessary.

File: pkg/vars/vars.go [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package vars
+
+import (
+	"time"
+
+	"github.com/pingcap/tiflow/cdc/model"
```
Comment (3AceShowHand) [inline pkg/vars/vars.go:19]: It looks so weired for one package in the `pkg` folder, reference some other package in the `cdc`

**Review by 3AceShowHand — APPROVED** [review]

**Review by asddongmen — APPROVED** [review]

File: pkg/vars/vars.go (hunk 3 above)
Comment (asddongmen) [reply pkg/vars/vars.go:19]: > It looks so weird for one package in the `pkg` folder, reference some other package in the `cdc`

Agreed!

Comment (sdojjy) [reply pkg/vars/vars.go:19]: moved to cdc package

File: pkg/vars/vars.go (hunk 2 above)
Comment (sdojjy) [reply pkg/vars/vars.go]: I will remove this in next PR

**Pull request merged**